### PR TITLE
mbsync: add module

### PR DESCRIPTION
**Pull request by pjones**
New service for [mbsync](http://isync.sourceforge.net/mbsync.html).

### CODE REVIEW

File: modules/services/mbsync.nix [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      description = "The package to use for the mbsync binary.";
+    };
+
+    frequency = mkOption {
+      type = types.str;
+      default = "*:0/5";
+      description = ''
+        How often to run mbsync.  This value is passed to the systemd
+        timer configuration as the onCalendar option.  See
+        systemd.time(7) for more information about the format.
```
Comment (rycee) [inline modules/services/mbsync.nix]: You can refer to man pages using a specific docbook markup (which admittedly is a bit clunky):
```xml
timer configuration as the onCalendar option.  See
systemd.time(7) for more information about the format.
```
becomes
```xml
timer configuration as the onCalendar option.  See
<citerefentry>
  <refentrytitle>systemd.time</refentrytitle>
  <manvolnum>7</manvolnum>
</citerefentry>
for more information about the format.
```

File: modules/services/mbsync.nix [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      '';
+    };
+
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Optional configuration file to link to use instead of
+        the default file (~/.mbsyncrc).
```
Comment (rycee) [inline modules/services/mbsync.nix]: More docbook :-)

```xml
the default file (~/.mbsyncrc).
```
should be
```xml
the default file (<filename>~/.mbsyncrc</filename>).
```

File: modules/services/mbsync.nix (hunk 1 above)
Comment (pjones) [reply modules/services/mbsync.nix]: Thanks for the tips.  Fixed.

**Comment by teto** [comment]
sorry I am a bit overwhelmed by work. This is the kind of program that I wish to be handled by my PR. As my PR is not ready yet (it works but the alot config creates errors with python2's config parser) and this PR links an existing config, it should be easy to integrate afterwards. Feel free to merge.

**Comment by rycee** [comment]
Thanks @teto for the feedback. I've rebased this to master in f26cc3b957ef77374f8b5275c1fc62217d13c6e3. Once we have a more general mail framework in place we can adjust the mbsync module as necessary.

Many thanks @pjones for the contribution :-)

**Pull request closed without merging**